components/Feed/StickyHeader.js

```diff
@@ -0,0 +1,78 @@
+import { Component } from 'react'
+import { HEADER_HEIGHT, HEADER_HEIGHT_MOBILE } from '../constants'
+import { css } from 'glamor'
+import { mediaQueries } from '@project-r/styleguide'
+
+const style = {
+  base: css({
+    padding: '8px 0',
+    borderTop: '1px solid #000',
+    position: 'relative',
+    backgroundColor: '#fff',
+    width: '100%'
+  }),
+  sticky: css({
+    position: 'fixed',
+    top: HEADER_HEIGHT - 1,
+    [mediaQueries.onlyS]: {
+      top: HEADER_HEIGHT_MOBILE - 1
+    }
+  })
+}
+
+class StickyHeader extends Component {
+  constructor (props) {
+    super(props)
+    this.originalOffset = null
+    this.state = {
+      sticky: false
+    }
+    this.ref = null
+    this.setRef = (el) => { this.ref = el }
+
+    this.onScroll = () => {
+      const y = window.pageYOffset
+      if (this.ref) {
+        const { sticky } = this.state
+        const currentOffset = this.ref.offsetTop
+        const nextSticky = y + this.getHeaderHeight() > (sticky ? this.originalOffset : currentOffset)
+        this.setState({ sticky: nextSticky })
+      }
+    }
+
+    this.getHeaderHeight = () => {
+      if (window.innerWidth >= mediaQueries.mBreakPoint) {
+        return HEADER_HEIGHT
+      } else {
+        return HEADER_HEIGHT_MOBILE
+      }
+    }
+  }
+
+  componentDidMount () {
+    this.originalOffset = this.ref.offsetTop
+    window.addEventListener('scroll', this.onScroll)
+  }
+
+  componentWillUnmount () {
+    window.removeEventListener('scroll', this.onScroll)
+  }
+
+  render () {
+    const { children } = this.props
+    const { sticky } = this.state
+    return (
+      <div
+        ref={this.setRef}
+        {...style.base}
+        className={sticky ? `${style.sticky}` : ''}
```

Comment:
This does not guarantee that `position: 'fixed'` wins. It's does happen because the base `css({...})` call is before the sticky `css({...})` call.

The situation as an static example:
https://codepen.io/anon/pen/GGEjrG

Two options to ensure it:
```js
{...merge(style.base, sticky && style.sticky)}
```
_Import merge from glamor. This comes at a slight runtime cost (hashing the css and injecting it the first time)._

Remove position from the glamor css and handle it with `style`:
```ks
{...style.base}
{...(sticky && style.sticky)}
style={{position: sticky ? 'fixed' : 'relative'}}
```
_This is faster at runtime, afaik._

Spreading `false`, `undefined` and `null` is safe:
```js
<div {...(false)} />
<div {...(undefined)} />
<div {...(null)} />
```
_Does nothing because object spread behaves like Object.assign_

Reply:
Let's handle it with the `style` prop – makes CSS loading efficient and the effect of the `sticky` flag explicit.